modules/src/mock/context.rs

```diff
@@ -365,7 +424,27 @@ impl MockContext {
         channel_end: ChannelEnd,
     ) -> Self {
         let mut channels = self.channels.clone();
-        channels.insert((port_id, chan_id), channel_end);
+        channels.insert((port_id.clone(), chan_id), channel_end);
+        let (module_id, _) = self
```

Comment:
Adding a channel automatically creates a new associated capability and is claimed by the module that owns the `port` capability.